File: 2025/C/cafeteria.c
#include <stdio.h>

int main(void) {
    int min_leite, max_leite, volume_copo, dose_cafe;
    scanf("%d\n%d\n%d\n%d", &min_leite, &max_leite, &volume_copo, &dose_cafe);
    
    int maximo_cafe = (volume_copo - min_leite);
    if (maximo_cafe % dose_cafe == 0) {
        printf("S\n");
        return 0;
    }
    
    int menor_quantidade_leite_possivel = min_leite + (maximo_cafe % dose_cafe);
    if (menor_quantidade_leite_possivel <= max_leite) {
        printf("S\n");
    } else {
        printf("N\n");
        return 0;
    }

    return 0;
}

File: 2025/python/cafeteria.py
min_leite = int(input())
max_leite = int(input())
volume_copo = int(input())
dose_cafe = int(input())

possivel_fazer = False
for volume_leite in range(min_leite, max_leite + 1):
    volume_cafe = volume_copo - volume_leite

    possivel_fazer = (volume_cafe % dose_cafe == 0)
    if possivel_fazer:
        break

print("S" if possivel_fazer else "N")